src/main/java/cinema/controller/OrderController.java

```diff
@@ -0,0 +1,48 @@
+package cinema.controller;
+
+import cinema.model.dto.order.OrderResponseDto;
+import cinema.service.OrderService;
+import cinema.service.ShoppingCartService;
+import cinema.service.UserService;
+import cinema.service.mapper.OrderMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+
+    @Autowired
+    public OrderController(OrderService orderService, UserService userService,
+                           ShoppingCartService shoppingCartService, OrderMapper orderMapper) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderMapper = orderMapper;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam("userId") Long userId) {
+        return orderMapper.mapToOrderResponseDto(
+                orderService.completeOrder(
+                        shoppingCartService.getByUser(userService.get(userId))));
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrderHistoryByUser(@RequestParam("userId") Long userId) {
```

Comment:
```suggestion
    public List<OrderResponseDto> getOrderHistoryByUser(@RequestParam Long userId) {
```
we can omit this everywhere